[V2] sgx_main: add sanity check when enumerating EPC resource

jiazhang0 opened the pull request:
V2:
sgx_main: clean up the redundant check during enumerating EPC resources
    
The start physical address of a EPC bank should not be zero if sgx is enabled.

V1 commit header:
According to the definition, the validity of EPC section start address
and size is determined by the lower 4 bits returned by EAX.

Currently, only 0 and 1 are defined values for the lower 4 bits of EAX
and ECX. So we add the sanity check for them in order to discover a
-NODEV returned if the valid lower bits of EAX and/or ECX extebded in
future.

Signed-off-by: Jia Zhang <qianyue.zj@alibaba-inc.com>

sergeay commented:
Main change here is not necessary.
Code before checks if sgx is enabled so no need to double check both eax and ecx for additional epc chunks.
Thanks.

jiazhang0 commented:
@sergeay Make sense. So these lines are redundant as well? 

```
diff --git a/sgx_main.c b/sgx_main.c
index a6d0b22..18c2a9a 100644
--- a/sgx_main.c
+++ b/sgx_main.c
@@ -224,8 +224,6 @@ static int sgx_init_platform(void)
                                (u64) (ecx & 0xfffff000);
                        sgx_epc_banks[sgx_nr_epc_banks].end =
                                sgx_epc_banks[sgx_nr_epc_banks].start + size;
-                       if (!sgx_epc_banks[sgx_nr_epc_banks].start)
-                               return -ENODEV;
                        sgx_nr_epc_banks++;
                } else {
                        break;
```

The start physical address should not be zero if sgx is enabled.

Commit:
sgx_main: clean up the redundant check during enumerating EPC resources

The start physical address of a EPC bank should not be zero if sgx is
enabled.

Signed-off-by: Jia Zhang <qianyue.zj@alibaba-inc.com>

sergeay commented:
Probably.
